plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

```diff
@@ -1274,7 +1274,45 @@ public Answer createVolumeFromSnapshot(final CopyCommand cmd) {
 
     @Override
     public Answer deleteSnapshot(final DeleteCommand cmd) {
-        return new Answer(cmd);
+        try {
+            SnapshotObjectTO snapshotTO = (SnapshotObjectTO) cmd.getData();
+            PrimaryDataStoreTO primaryStore = (PrimaryDataStoreTO) snapshotTO.getDataStore();
+            VolumeObjectTO volume = snapshotTO.getVolume();
+            String snapshotFullPath = snapshotTO.getPath();
+            String snapshotName = snapshotFullPath.substring(snapshotFullPath.lastIndexOf("/") + 1);
+            KVMStoragePool primaryPool = storagePoolMgr.getStoragePool(primaryStore.getPoolType(), primaryStore.getUuid());
+            KVMPhysicalDisk disk = storagePoolMgr.getPhysicalDisk(primaryStore.getPoolType(), primaryStore.getUuid(), volume.getPath());
+            if (primaryPool.getType() == StoragePoolType.RBD) {
+                Rados r = new Rados(primaryPool.getAuthUserName());
+                r.confSet("mon_host", primaryPool.getSourceHost() + ":" + primaryPool.getSourcePort());
+                r.confSet("key", primaryPool.getAuthSecret());
+                r.confSet("client_mount_timeout", "30");
+                r.connect();
+                s_logger.debug("Succesfully connected to Ceph cluster at " + r.confGet("mon_host"));
+                IoCTX io = r.ioCtxCreate(primaryPool.getSourceDir());
+                Rbd rbd = new Rbd(io);
+                RbdImage image = rbd.open(disk.getName());
+                try {
+                    s_logger.info("Attempting to remove RBD snapshot " + disk.getName() + "@" + snapshotName);
+                    if (image.snapIsProtected(snapshotName)) {
+                        s_logger.debug("Unprotecting snapshot " + snapshotFullPath);
+                        image.snapUnprotect(snapshotName);
+                    }
+                    image.snapRemove(snapshotName);
+                    s_logger.info("Snapshot " + snapshotFullPath + " successfully removed.");
+                } finally {
+                    rbd.close(image);
+                    r.ioCtxDestroy(io);
+                }
+            } else {
+                s_logger.warn("Operation not implemented!");
+                throw new InternalErrorException("Operation not implemented!");
```

Comment:
Wido, what do your mean? Improve s_logger.info messages? And include shared/dedicated message for pool type? Because if you mean RBD pool type - it's already included into message " .. remove RBD sn..."